lib/src/Uncertainty/Algorithm/Simulation/QuantileConfidence.cxx

```diff
@@ -0,0 +1,352 @@
+  // Nothing to to do
+}
+
+/* Constructor with parameters */
+QuantileConfidence::QuantileConfidence(const Scalar alpha, const Scalar beta)
+  : PersistentObject()
+  , alpha_(alpha)
+  , beta_(beta)
+{
+  if (!(alpha >= 0.0) || !(alpha <= 1.0))
+    throw InvalidArgumentException(HERE) << "Quantile level must be in [0, 1]";
+  if (!(beta >= 0.0) || !(beta <= 1.0))
+    throw InvalidArgumentException(HERE) << "Confidence level must be in [0, 1]";
+}
+
+/* Virtual constructor */
+QuantileConfidence * QuantileConfidence::clone() const
+{
+  return new QuantileConfidence(*this);
+}
+
+/* String converter */
+String QuantileConfidence::__repr__() const
+{
+  return OSS() << GetClassName()
+         << " alpha=" << alpha_
+         << " beta=" << beta_;
+}
+
+String QuantileConfidence::__str__(const String & /*offset*/) const
+{
+  return OSS() << GetClassName() << "(alpha=" << alpha_ << ", beta=" << beta_ << ")";
+}
+
+UnsignedInteger QuantileConfidence::computeUnilateralRank(const UnsignedInteger size, const Bool tail) const
+{
+  const UnsignedInteger minimumSize = computeUnilateralMinimumSampleSize(0, tail);
+  if (size < minimumSize)
+    throw InvalidArgumentException(HERE) << "Cannot compute unilateral rank as size (" << size << ") is lower than minimum size (" << minimumSize << ")";
+
+  const Binomial binomial(size, alpha_);
+  const Scalar p = tail ? std::pow(1.0 - alpha_, 1.0 * size) : 1.0 - std::pow(alpha_, 1.0 * size);
+
+  if (tail && (p > 1.0 - beta_))
+    throw InvalidArgumentException(HERE) << "Cannot compute rank as parameters do not satisfy (1 - alpha)^n >= 1 - beta";
+  if (!tail && (p < beta_))
+    throw InvalidArgumentException(HERE) << "Cannot compute rank as parameters do not satisfy 1 - alpha^n >= beta";
+
+  const UnsignedInteger rank = binomial.computeQuantile(beta_, !tail)[0];
+  return std::min(rank, size - 1);
+}
+
+
+Indices QuantileConfidence::computeBilateralRank(const UnsignedInteger size) const
+{
+  const UnsignedInteger minimumSize = computeBilateralMinimumSampleSize();
+  if (size < minimumSize)
+    throw InvalidArgumentException(HERE) << "Cannot compute bilateral rank as size (" << size << ") is lower than minimum size (" << minimumSize << ")";
+
+  // find the indices of interval ]k1, k2] with smallest probability >=beta according to a Binomial(n=size, alpha)
+  const Binomial binomial(size, alpha_);
+  Scalar pBest = SpecFunc::MaxScalar;
+  UnsignedInteger k1Best = size;
+  UnsignedInteger k2Best = size;
+  Scalar p1Prev = 0.0;
+  UnsignedInteger k1 = 0;
+  while (k1 < size)
+  {
+    // find upper bound of cdf increment location
+    Scalar p1 = binomial.computeCDF(k1);
+    UnsignedInteger inc = 1;
+    UnsignedInteger k1Prev = k1;
+    while (p1 == p1Prev)
+    {
+      k1Prev = k1;
+      k1 += inc;
+      p1 = binomial.computeCDF(k1);
+      inc *= 2;
+    }
+
+    // find exact cdf increment location
+    UnsignedInteger k1Max = k1;
+    UnsignedInteger k1Min = k1Prev;
+    while (k1Max - k1Min > 1)
```

Comment:
According to CodeCov, this part of the code is never tested:

![image](https://github.com/user-attachments/assets/b2a5f707-1425-4c0d-9c41-4fe4247f1dc9)

Can you find an example where this part of the algorithm is activated?